repos/fdbt-site/src/pages/api/selectSalesOfferPackage.ts

```diff
@@ -71,9 +79,17 @@ export const sanitiseReqBody = (
 export default (req: NextApiRequestWithSession, res: NextApiResponse): void => {
     try {
         const multipleProductAttribute = getSessionAttribute(req, MULTIPLE_PRODUCT_ATTRIBUTE);
-        const products = multipleProductAttribute
-            ? multipleProductAttribute.products
-            : [{ productName: 'product', productPrice: '' }];
+        const ticket = getSessionAttribute(req, MATCHING_JSON_ATTRIBUTE);
+        const matchingJsonMetaData = getSessionAttribute(req, MATCHING_JSON_META_DATA_ATTRIBUTE);
+
+        let prod = [];
+        if (ticket?.products && 'productName' in ticket.products[0]) {
+            prod = [{ productName: ticket.products[0].productName, productPrice: ticket.products[0].productPrice }];
+        } else {
+            prod = [{ productName: 'product', productPrice: '' }];
+        }
+
+        const products = multipleProductAttribute ? multipleProductAttribute.products : prod;
```

Comment:
On line 85, we could instead do `const products = multipleProductAttribute ? multipleProductAttribute.products : [];`